versions/3.1.0.md

```diff
@@ -2294,46 +2299,50 @@ The following properties are taken directly from the JSON Schema definition and
 - uniqueItems
 - maxProperties
 - minProperties
-- required
 - enum
-
-The following properties are taken from the JSON Schema definition but their definitions were adjusted to the OpenAPI Specification. 
-- type - Value MUST be a string. Multiple types via an array are not supported.
-- allOf - Inline or referenced schema MUST be of a [Schema Object](#schemaObject) and not a standard JSON Schema.
-- oneOf - Inline or referenced schema MUST be of a [Schema Object](#schemaObject) and not a standard JSON Schema.
-- anyOf - Inline or referenced schema MUST be of a [Schema Object](#schemaObject) and not a standard JSON Schema.
-- not - Inline or referenced schema MUST be of a [Schema Object](#schemaObject) and not a standard JSON Schema.
-- items - Value MUST be an object and not an array. Inline or referenced schema MUST be of a [Schema Object](#schemaObject) and not a standard JSON Schema. `items` MUST be present if the `type` is `array`.
-- properties - Property definitions MUST be a [Schema Object](#schemaObject) and not a standard JSON Schema (inline or referenced).
-- additionalProperties - Value can be boolean or object. Inline or referenced schema MUST be of a [Schema Object](#schemaObject) and not a standard JSON Schema. Consistent with JSON Schema, `additionalProperties` defaults to `true`.
+- propertyNames
+- contains
+- const
+- examples
+- if / then / else
```

Comment:
Worried that for instance these keys will make creating tooling which supports 3.1 much difficult than supporting 3.0

Reply:
@jcarres-mdsol could you quantify that in any way? What sort of tooling are you concerned about, and how do you think it is likely to make tooling more complicated? Is it simply "more is worse" or are there specific concerns about if / then / else in particular? 

Did you see my link to the notes about how `if: X, then: Y, else: Z` is equivalent to `oneOf: [allOf: [X, Y], allOf: [not: X, Z]]`?

Reply:
@jcarres-mdsol the other important thing about moving to draft-08 is that it supports extensible vocabularies, and one of the primary use cases for that is to improve OpenAPI tooling support.

For example, it is currently very awkward to use JSON Schema as a data definition language- it is simply not designed for that.  The current JSON Schema vocabulary is a constraint system, not a definition system.

However, draft-08 will allow defining keywords that (for example) can disambiguate how to interpret an `allOf` (is this an inheritance structure where on branch is the base class and therefore code gen tools should generate and connect two classes? or is it just gluing some re-usable fields together and the code-gen tool should ignore the pieces and generate one class from the result of the `allOf`?)

Ultimately, draft-08 should make tooling **dramatically easier** to create.  That's the whole point of it.  But those parts don't show up in this PR yet for a variety of reasons.

Reply:
From a validation perspective, `if`/`then`/`else` says how to figure out which set of rules to use, and how to apply them.

From a data definition perspective, an `if`/`then`/`else` probably means something like "there are two subclasses here."

We need to stop trying to contort generative tooling (code gen, ui gen, doc gen) to fit a constraint system and instead add keywords that support generative use cases.  Using the JSON Schema validation vocabulary to generate things is like trying to hammer a nail with a screwdriver.  You can probably make it happen if the screwdriver has a nice heavy handle, but it's kind of ridiculous to do.

Up until now, OpenAPI has only had the options of doing that hammering, or breaking compatibility with JSON Schema.  And they've tried to strike a balance between those two distasteful options.  Draft-08 is designed to move past that problem and provide a real solution.

Reply:
I was thinking of a couple of use-cases. 
One is generation for statically typed languages but as per your explanation of the meaning of `if-then-else`, if generator already have support for `oneOf` we have already crossed that bridge so I would rectify my comment, it should not be much extra work.

in-house we have a linter which applies some rules, we have had multiple fixes over `oneOf` , `allOf`. I do agree, bringing this vocabulary more in line with draft-8 would be at least be less confusing for people implementing the tools. 

So, I'd like to change my comment to a +1 on this!